popgen/MCMC_Selection.go

```diff
@@ -82,15 +84,25 @@ func InitializeTheta(m float64, s float64, k int) Theta {
 
 //MetropolisHastings implements the MH algorithm for Markov Chain Monte Carlo approximation of the posterior distribution for selection based on an input allele frequency spectrum.
 //muZero and sigmaZero represent the starting hyperparameter values.
-func MetropolisHastings(data AFS, muZero float64, sigmaZero float64, iterations int, logSpace bool) ([]float64, []float64, []bool) {
+func MetropolisHastings(data AFS, muZero float64, sigmaZero float64, iterations int) ([]float64, []float64, []bool) {
+	//profiling test code
+	f, err := os.Create("testProfile.prof")
+	if err != nil {
+		log.Fatal(err)
+	}
+	pprof.StartCPUProfile(f)
+	defer pprof.StopCPUProfile()
+
+	binomMap := make(map[int][]float64)
+
 	muList := make([]float64, iterations)
 	sigmaList := make([]float64, iterations)
 	acceptList := make([]bool, iterations)
 	//initialization to uninformative standard normal
 	t := InitializeTheta(muZero, sigmaZero, len(data.sites))
 	for i := 0; i < iterations; i++ {
 		tCandidate := GenerateCandidateThetaPrime(t)
-		if Metropolis_Accept(t, tCandidate, data, logSpace) {
+		if Metropolis_Accept(t, tCandidate, data, &binomMap) {
```

Comment:
I am interested in the choice of using a pointer to a map.  I think that maps (like slices) have their underlying data passed by reference, with only a header-like collection of data being copied.  I am not asking for a change, just interested to hear your thoughts.